pkg/bootstrap/docker/up.go

```diff
@@ -583,7 +596,14 @@ func (c *ClientStartConfig) importObjects(out io.Writer, locations map[string]st
 }
 
 func (c *ClientStartConfig) openShiftImage() string {
-	return fmt.Sprintf("openshift/origin:%s", c.Tag)
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = c.Images
+	image, err := imageTemplate.Expand("")
+	if err != nil {
+		// should not happen because template was validated
+		panic(err)
+	}
+	return strings.Replace(image, "origin-:", "origin:", -1)
```

Comment:
I don't like that either.  Hrm.

On Wed, May 18, 2016 at 3:37 PM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/bootstrap/docker/up.go
> <https://github.com/openshift/origin/pull/8932#discussion_r63767758>:
>
> > @@ -583,7 +596,14 @@ func (c *ClientStartConfig) importObjects(out io.Writer, locations map[string]st
> >  }
> >
> >  func (c *ClientStartConfig) openShiftImage() string {
> > -	return fmt.Sprintf("openshift/origin:%s", c.Tag)
> > +	imageTemplate := variable.NewDefaultImageTemplate()
> > +	imageTemplate.Format = c.Images
> > +	image, err := imageTemplate.Expand("")
> > +	if err != nil {
> > +		// should not happen because template was validated
> > +		panic(err)
> > +	}
> > +	return strings.Replace(image, "origin-:", "origin:", -1)
>
> yuck... don't see a nice way to do that with the current implementation,
>
> https://github.com/openshift/origin/blob/master/pkg/cmd/util/variable/imagetemplate.go#L45
>
> How about using a marker as a coponent name (##REPLACE_ME##) and replace
> both ("-"+marker) and (marker) with blank?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8932/files/c257f2dddd468b79daead009fddfb58cd2ef165f#r63767758>
>